microsoft/testsuites/vm_extensions/kvvm_extension.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+import random
+
+from assertpy import assert_that
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.compute import ComputeManagementClient  # type: ignore
+from azure.mgmt.keyvault import KeyVaultManagementClient
+from azure.mgmt.keyvault.models import (
+    AccessPolicyEntry,
+    Permissions,
+    Sku,
+    VaultCreateOrUpdateParameters,
+    VaultProperties,
+)
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.operating_system import FreeBSD
+from lisa.sut_orchestrator import AZURE
+from lisa.sut_orchestrator.azure.common import (
+    AzureNodeSchema,
+    check_system_status,
+    create_certificates,
+    get_node_context,
+    rotate_certificates,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.testsuite import TestResult
+from lisa.util import SkippedException
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="BVT for Azure Key Vault Extension",
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class AzureKeyVaultExtensionBvt(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        The following test case validates the Azure Key Vault Linux
+        Extension while creating the following resources:
+        A resource group
+        A VM
+        A Key Vault
+        Two certificates in the Key Vault
+        Retrieval of the certificate's secrets
+        through SecretClient class from Azure SDK.
+        Installation of the Azure Key Vault Linux Extension on the VM.
+        Rotation of the certificates (After KVVM Extension has been installed)
+        All of the resources have been created by using the Azure SDK Python.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_key_vault_extension(
+        self, log: Logger, node: Node, result: TestResult
+    ) -> None:
+        # Section for vault name and supported OS check
+        vault_name = os.getenv(
+            "vault_name", f"python-keyvault-{random.randint(1, 100000):05}"
+        )
+        vault_name_a = vault_name + "prp"
+        if isinstance(node.os, FreeBSD):
+            raise SkippedException(f"unsupported distro type: {type(node.os)}")
+
+        # Section for environment setup
+        environment = result.environment
```

Comment:
add a method in azure/common.py to enable SYSTEM_ASSIGNED_IDENTITY for the vm.
refer https://learn.microsoft.com/en-us/samples/azure-samples/compute-python-msi-vm/compute-python-msi-vm/